> @vjosadika Please provide previous behavior to confirm this is not a backwards incompatible change.

@CuriousVini The previous behavior of DB Sink:
1. Was throwing Exception when we were trying to insert a Decimal value to a non-nullable field in DB.
```
PostgreSQL - old sink using decimal:
java.io.IOException: org.postgresql.util.PSQLException: No value specified for parameter 4.

MySQL - old sink using decimal:
Caused by: java.sql.SQLException: Field 'decimal' doesn't have a default value
```
2. Was storing null value for Decimal values when the field was nullable in DB.

The above two scenarios were happening because, for logical types, there was no case for DECIMAL type and the value for decimal column in the statement was always null (L294).

https://github.com/cdapio/hydrator-plugins/blob/9073013b7d41aee8fbe57df9d70e10efc6bf4209/database-plugins/src/main/java/io/cdap/plugin/DBRecord.java#L269-L295

--- a/database-plugins/src/main/java/io/cdap/plugin/DBRecord.java
+++ b/database-plugins/src/main/java/io/cdap/plugin/DBRecord.java
@@ -285,6 +285,11 @@ public class DBRecord implements Writable, DBWritable, Configurable, DataSizeRep
           stmt.setTimestamp(sqlIndex, Timestamp.from(record.getTimestamp(fieldName).toInstant()));
           bytesWritten += Long.BYTES;
           break;
+        case DECIMAL:
+          BigDecimal value = record.getDecimal(fieldName);
+          stmt.setBigDecimal(sqlIndex, value);
+          bytesWritten += value.unscaledValue().bitLength() / Byte.SIZE + Integer.BYTES;
+          break;
       }
       return;
     }

--- a/database-plugins/src/test/java/io/cdap/plugin/DatabasePluginTestBase.java
+++ b/database-plugins/src/test/java/io/cdap/plugin/DatabasePluginTestBase.java
@@ -194,7 +194,9 @@ public class DatabasePluginTestBase extends HydratorTestBase {
                      "DECIMAL_INT DECIMAL(8, 0)," +
                      "DECIMAL_LONG DECIMAL(11, 0)," +
                      "NUMERIC_INT NUMERIC(8, 0)," +
-                     "NUMERIC_LONG NUMERIC(11, 0)" +
+                     "NUMERIC_LONG NUMERIC(11, 0)," +
+                     "DECIMAL_DOUBLE DECIMAL(10, 2)," +
+                     "NUMERIC_DOUBLE DECIMAL(10, 2)" +
                      ")");
       stmt.execute("CREATE TABLE \"MY_DEST_TABLE\" AS (" +
                      "SELECT * FROM \"my_table\") WITH DATA");
@@ -208,11 +210,11 @@ public class DatabasePluginTestBase extends HydratorTestBase {
       PreparedStatement pStmt1 =
         conn.prepareStatement("INSERT INTO \"my_table\" " +
                                 "VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, " +
-                                "?, ?)");
+                                "?, ?, ?, ?)");
       PreparedStatement pStmt2 =
         conn.prepareStatement("INSERT INTO \"your_table\" " +
                                 "VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, " +
-                                "?, ?)")) {
+                                "?, ?, ?, ?)")) {
       // insert the same data into both tables: my_table and your_table
       final PreparedStatement[] preparedStatements = {pStmt1, pStmt2};
       for (PreparedStatement pStmt : preparedStatements) {
@@ -249,6 +251,8 @@ public class DatabasePluginTestBase extends HydratorTestBase {
           pStmt.setLong(25, 999999999L + i);
           pStmt.setInt(26, 123 + i);
           pStmt.setLong(27, 999999999L + i);
+          pStmt.setDouble(28, 0);
+          pStmt.setDouble(29, 0);
           pStmt.executeUpdate();
         }
       }

--- a/database-plugins/src/test/java/io/cdap/plugin/db/batch/sink/DBSinkTestRun.java
+++ b/database-plugins/src/test/java/io/cdap/plugin/db/batch/sink/DBSinkTestRun.java
@@ -35,6 +35,7 @@ import io.cdap.plugin.db.batch.source.DBSource;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.math.BigDecimal;
 import java.sql.Connection;
 import java.sql.Date;
 import java.sql.PreparedStatement;
@@ -62,7 +63,7 @@ public class DBSinkTestRun extends DatabasePluginTestBase {
     String inputDatasetName = "input-dbsinktest";
 
     String cols = "ID, NAME, SCORE, GRADUATED, TINY, SMALL, BIG, FLOAT_COL, REAL_COL, NUMERIC_COL, DECIMAL_COL, " +
-      "BIT_COL, DATE_COL, TIME_COL, TIMESTAMP_COL, BINARY_COL, BLOB_COL, CLOB_COL";
+      "BIT_COL, DATE_COL, TIME_COL, TIMESTAMP_COL, BINARY_COL, BLOB_COL, CLOB_COL, NUMERIC_DOUBLE, DECIMAL_DOUBLE";
     ETLPlugin sourceConfig = MockSource.getPlugin(inputDatasetName);
     ETLPlugin sinkConfig = new ETLPlugin("Database",
                                          BatchSink.PLUGIN_TYPE,
@@ -145,8 +146,10 @@ public class DBSinkTestRun extends DatabasePluginTestBase {
       Schema.Field.of("BIG", Schema.of(Schema.Type.LONG)),
       Schema.Field.of("FLOAT_COL", Schema.of(Schema.Type.FLOAT)),
       Schema.Field.of("REAL_COL", Schema.of(Schema.Type.FLOAT)),
-      Schema.Field.of("NUMERIC_COL", Schema.of(Schema.Type.DOUBLE)),
-      Schema.Field.of("DECIMAL_COL", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("NUMERIC_COL", Schema.decimalOf(10, 2)),
+      Schema.Field.of("NUMERIC_DOUBLE", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("DECIMAL_COL", Schema.decimalOf(10, 2)),
+      Schema.Field.of("DECIMAL_DOUBLE", Schema.of(Schema.Type.DOUBLE)),
       Schema.Field.of("BIT_COL", Schema.of(Schema.Type.BOOLEAN)),
       Schema.Field.of("DATE_COL", Schema.of(Schema.LogicalType.DATE)),
       Schema.Field.of("TIME_COL", Schema.of(Schema.LogicalType.TIME_MICROS)),
@@ -169,8 +172,10 @@ public class DBSinkTestRun extends DatabasePluginTestBase {
                          .set("BIG", 3456987L)
                          .set("FLOAT_COL", 3.456f)
                          .set("REAL_COL", 3.457f)
-                         .set("NUMERIC_COL", 3.458d)
-                         .set("DECIMAL_COL", 3.459d)
+                         .set("NUMERIC_DOUBLE", 3.458d)
+                         .setDecimal("NUMERIC_COL", BigDecimal.valueOf(6.45d))
+                         .set("DECIMAL_DOUBLE", 3.459d)
+                         .setDecimal("DECIMAL_COL", BigDecimal.valueOf(6.46d))
                          .set("BIT_COL", (i % 2 == 1))
                          .setDate("DATE_COL", localDateTime.toLocalDate())
                          .setTime("TIME_COL", localDateTime.toLocalTime())

